crates/ironrdp-cliprdr-format/src/html.rs

```diff
@@ -45,53 +54,44 @@ pub fn cf_html_to_plain_html(input: &[u8]) -> Result<String, HtmlError> {
-                    // We reached the end of the headers, but we didn't find the required ones,
-                    // so the format is invalid.
+                // At this point, we reached the end of the headers.
+                if let (Some(start), Some(end)) = (start_fragment, end_fragment) {
+                    let start = usize::try_from(start).map_err(|_| HtmlError::InvalidConversion)?;
+                    let end = usize::try_from(end).map_err(|_| HtmlError::InvalidConversion)?;
+
+                    // Ensure start and end values are properly bounded.
+                    if start > end || end > input.len() {
+                        return Err(HtmlError::InvalidFormat);
+                    }
+
+                    // Extract the fragment from the original buffer.
+                    let fragment = core::str::from_utf8(&input[start..end])?;
+
+                    return Ok(fragment);
+                } else {
+                    // If required headers were not found, the input is considered invalid.
                     return Err(HtmlError::InvalidFormat);
                 }
             }
         };
 
-        if let (Some(start), Some(end)) = (start_fragment, end_fragment) {
-            let start = usize::try_from(start).map_err(|_| HtmlError::InvalidConversion)?;
-            let end = usize::try_from(end).map_err(|_| HtmlError::InvalidConversion)?;
-
-            // Extract fragment from the original buffer.
-            if start > end || end > input.len() {
-                return Err(HtmlError::InvalidFormat);
-            }
-
-            break String::from_utf8(input[start..end].to_vec())?;
-        }
-
-        // INVARIANT: end_pos < headers_cursor.len() - 1
-        // This is safe because we already checked above that the line ends with `\r` or `\n`.
-        #[allow(clippy::arithmetic_side_effects)]
-        {
-            // Go to the next line, skipping any leftover `LF` if CRLF was used.
-            let has_leftover_lf = end_pos + 1 != headers_cursor.len()
-                && headers_cursor[end_pos] == b'\r'
-                && headers_cursor[end_pos + 1] == b'\n';
-
-            if has_leftover_lf {
-                headers_cursor = &headers_cursor[end_pos + 2..];
-            } else {
-                headers_cursor = &headers_cursor[end_pos + 1..];
-            }
+        // Skip EOL control characters and prepare for next line.
+        cursor = &cursor[line.len()..];
+        while let Some(b'\n' | b'\r') = cursor.first() {
+            cursor = &cursor[1..];
         }
-    };
-
-    return Ok(fragment);
+    }
 
     fn header_value_to_u32(value: &str) -> Result<u32, std::num::ParseIntError> {
         value.trim_start_matches('0').parse::<u32>()
     }
 }
 
 /// Converts plain HTML text to `CF_HTML` format.
-pub fn plain_html_to_cf_html(fragment: &str) -> Vec<u8> {
-    let mut buffer = Vec::new();
+pub fn plain_html_to_cf_html(fragment: &str) -> String {
```

Comment:
Since the `CF_HTML` format is using UTF-8, we can encode that in the output type and return `String`.